src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,231 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC. All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "dbus"
+
+module Yast
+  class SnapshotDBus
+    extend Forwardable
+    include Yast::Logger
+
+    SNAPSHOT_ID = {
+      0 => "single",
+      1 => "pre",
+      2 => "post"
+    }.freeze
+
+    def_delegators :@context, :config, :number, :pre_number
+
+    attr_accessor :context
+
+    def initialize(snapshot = nil)
+      @context = snapshot
+    end
+
+    def all(configs)
+      configs ||= list_configs
```

Comment:
here maybe on other hand make sense to have configs with nil default as it is initialized here if not passed anything